### PR TITLE
Disable closing of the httptest Server due to https://github.com/golang/go/issues/12262

### PR DESCRIPTION
**Pull request by brendandburns**
@davidopp @wojtek-t @ihmccreery 

Ref https://github.com/kubernetes/kubernetes/issues/18928

### CODE REVIEW

**Comment by ihmccreery** [comment]
Also xref #19458 as precedent.

**Comment by dchen1107** [comment]
This PR is for unittest, and travis passed. I am manually merging it now without waiting for gce e2e.

File: pkg/storage/etcd/testing/utils.go [hunk 1]
```diff
@@ -155,7 +155,8 @@ func (m *EtcdTestServer) Terminate(t *testing.T) {
 	time.Sleep(250 * time.Millisecond)
 	for _, hs := range m.hss {
 		hs.CloseClientConnections()
-		hs.Close()
+		// TODO: Uncomment when fix #19254
+		// hs.Close()
```
Comment (timothysc) [inline pkg/storage/etcd/testing/utils.go]: umm so we now have 2 band-aides...

/cc @xiang90

Comment (brendandburns) [inline pkg/storage/etcd/testing/utils.go]: Yeah, there was broad band-aid put in in https://github.com/kubernetes/kubernetes/pull/19458 at least this is consistent with that?

**Comment by ihmccreery** [comment]
> This PR is for unittest, and travis passed. I am manually merging it now without waiting for gce e2e.

@dchen1107 As far as I know, Travis doesn't run unit tests.  @spxtr

**Pull request merged**